managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1667,9 +1674,7 @@ void createNewOpAddEntryForNewLedger() {
             if (existsOp != null) {
                 // If op is used by another ledger handle, we need to close it and create a new one
                 if (existsOp.ledger != null) {
-                    existsOp.close();
-                    existsOp = OpAddEntry.createNoRetainBuffer(existsOp.ml, existsOp.data,
-                            existsOp.getNumberOfMessages(), existsOp.callback, existsOp.ctx);
+                    existsOp = existsOp.duplicateAndClose(currentLedgerTimeoutTriggered);
```

Comment:
Not related to this change, but how does the reference count (`retain`/`release`) of the `ByteBuf data` in the duplicated op get handled?

Reply:
The `Bytebuf data` will be released after the entry is added completely or the Managed Ledger is closed. see:
- `OpAddEntry.addComplete` -> `OpAddEntry.run`
- `ML.clearPendingAddEntries` -> `OpAddEntry.failed`